docs/host_install.md

```diff
@@ -0,0 +1,189 @@
+building from source are available
+[Intel SGX SDK git repository](https://github.com/intel/linux-sgx).
+
+## Build and Install SGX SSL
+
+SGX OpenSSL is a compilation of OpenSSL specifically for use within SGX
+enclaves. We have tested PDO with SGX SSL version 2.4.1.
+
+Detailed instructions for building and installing SGX SSL is available
+from the
+[Intel SGX SSL github repository](https://github.com/intel/intel-sgx-ssl).
+
+Follow these steps to compile and install version 2.4.1:
+
+- Ensure you have the SGX SDK environment variables activated:
+```bash
+source /opt/intel/sgxsdk/environment
+```
+
+- Clone the SGX SSL source:
+```bash
+git clone 'https://github.com/intel/intel-sgx-ssl.git'
+```
+
+- Check out the recommended version (v2.4.1):
+
+```bash
+cd intel-sgx-ssl
+git checkout v2.4.1
+```
+
+- Download the OpenSSL source package that will form the base of this
+SGX SSL install:
+
+```bash
+cd openssl_source
+wget 'https://www.openssl.org/source/openssl-1.1.0j.tar.gz'
+cd ..
+```
+
+- Set the environment variable for hardware or simulation mode. For
+simulation mode use `export SGX_MODE=SIM`. For hardware mode use `export
+SGX_MODE=HW`.
+
+- Compile and install the SGX SSL project.
+```bash
+cd Linux
+make DESTDIR=/opt/intel/sgxssl all test
+sudo make install
+```
+
+- Export the `SGX_SSL` environment variable to enable the build
+utilities to find and link this library.  Consider adding this to your
+login shell script (`~/.bashrc` or similar)
+
+```bash
+export SGX_SSL=/opt/intel/sgxssl
+```
+
+## Build and Install Sawtooth
+
+Hyperledger Private Data Objects uses the Hyperledger Sawtooth distributed
+ledger to store data object instances and state, and to guarantee update
+atomicity.
+
+Application logic is implemented in Sawtooth through the use of Transaction
+Processors; transaction processors enable the distributed ledger to handle
+application requests. This repository contains the code required to build
+Transaction Processors that handle PDO requests.
+
+Follow the
+[setup document](sawtooth/docs/SETUP.md)
+to install both Sawtooth and the custom Sawtooth Transaction Processors.
+
+Note that the Sawtooth components do not depend on any other components of the
+PDO project, and can be set up on an entirely separate machine from the one
+running Private Data Objects. It is recommended that Sawtooth be run on Ubuntu
+16.04 as it is the only operating system version on which Sawtooth is actively
+supported.
+
+## Build and Install PDO
+
+Assuming you have installed and configured the pre-requisites in the
+default location lcations, the following commands will build and install
+PDO into a Python virtual environment in the directory
+`${PDO_SOURCE_ROOT}/build/_dev`.
+
+```bash
+export TINY_SCHEME_SRC=${PDO_SOURCE_ROOT}/tinyscheme-1.41
+export SGX_MODE=SIM
+export SGX_SSL=/opt/intel/sgxssl
+source /opt/intel/sgxsdk/environment
+source ${PDO_SOURCE_ROOT}/build/common-config.sh
+
+cd ${PDO_SOURCE_ROOT}/build
+make
+make test
+```
```

Comment:
it would be great if all the commands above can be part of the make file and we have a command like "make install". Given the absence of user input, and given that at this point, the user is also executing commands mechanically, it might make sense to automate everything rather than semi automate.